### DISCUSSION TITLE
Is it possible to use an external authentication system instead of the native one that Rotki uses?

### DISCUSSION THREAD

**Discussion by karvounis** [rotki/rotki#4388] [Q&A] [answered]
Hello there,

I am currently running Rotki on Docker Swarm, sitting behind Traefik. I started offloading the authentication process to [Authelia](https://www.authelia.com/) + OpenLDAP and I am trying to migrate all the services to use this authentication mechanism.This allows you to have 1 user for all services and use Authelia as SSO. I wanted to ask if Rotki allows that. If it is possible to offload the authentication mechanism to something external like Authelia and be able to login as a Rotki user.
Bonus Q: What happens to your already created users through Rotki and their configuration (Exchange keys etc)?

Cheers!

**Comment by kelsos** [comment, marked as answer]
Hey @karvounis the setup sounds interesting but I can say for certain that at the moment rotki doesn't support any external authentication. 

Unfortunately without any deeper knowledge of the systems involved, I cannot say if this would be possible, or how much effort would it require.

**Reply by karvounis** [reply]
Hey @kelsos thanks for the response!  The added value is that Rotki, like many other software, could leverage tools that their main focus is authentication/authorisation.
I wonder if it is possible to add this request to your backlog with low priority? I would gladly help with it and I think it would be a nice to-have.

**Reply by kelsos** [reply]
I am not sure how much effort that would require on our side, but if we could have a specification on what is needed I don't see why not.

**Reply by karvounis** [reply]
Who do you think can provide the specification for this?

**Reply by kelsos** [reply]
Not sure, I could potentially do it if I can get more information on how such an external authentication system would interact with rotki, but I am at a loss on where to start from.

**Reply by karvounis** [reply]
You can start with how to authenticate using [LDAP](https://www.okta.com/identity-101/what-is-ldap/), [OpenLDAP](https://www.openldap.org/) (the one I am most interested about :P ), [OIDC](https://goteleport.com/blog/how-oidc-authentication-works/), [Github OAuth](https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps).

#### How others do it

- https://docs.min.io/minio/baremetal/security/iam-overview.html
- https://docs.portainer.io/admin/settings/authentication/ldap
- https://docs.firefly-iii.org/firefly-iii/advanced-installation/authentication
- https://vikunja.io/docs/config-options/#auth

#### Useful software to check

- https://www.authelia.com/
- https://www.openldap.org/

I am available for any question you might have!

**Reply by kelsos** [reply]
Thank I will have a look when I manage to find some time
